llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -5038,12 +5044,32 @@ InstCombinerImpl::foldICmpWithMinMaxImpl(Instruction &I,
         NewPred = ICmpInst::getInversePredicate(NewPred);
       return ICmpInst::Create(Instruction::ICmp, NewPred, X, Y);
     }
-    // Otherwise (X != Z, nofold):
-    //       Expr      Result
-    // min(X, Y) == Z X > Y || Y == Z
-    // max(X, Y) == Z X < Y || Y == Z
-    // min(X, Y) != Z X <= Y && Y != Z
-    // max(X, Y) != Z X >= Y && Y != Z
+    // Otherwise (X != Z):
+    ICmpInst::Predicate NewPred = MinMax->getPredicate();
+    auto NewCmpXZ = IsCondKnownTrue(simplifyICmpInst(NewPred, X, Z, Q));
+    if (!NewCmpXZ.has_value()) {
+      std::swap(X, Y);
+      std::swap(CmpXZ, CmpYZ);
+      NewCmpXZ = IsCondKnownTrue(simplifyICmpInst(NewPred, X, Z, Q));
+    }
```

Comment:
This is also applicable to other predicates no?
I.e
`max(X, Y) u<= Z`
If you have `MinMax`(`UGT`) known true -> true. Known false -> `Y u<= Z`.

Reply:
It has been addressed by [D156238](https://reviews.llvm.org/D156238).

https://github.com/llvm/llvm-project/blob/07151f0241d3f893cb36eb2dbc395d4098f74a87/llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp#L5064-L5097